File: 4VODS_API/src/Entity/Diffusion.php
<?php

namespace App\Entity;

use App\Repository\DiffusionRepository;
use Doctrine\ORM\Mapping as ORM;
use App\Entity\Iniciative;

#[ORM\Entity(repositoryClass: DiffusionRepository::class)]
class Diffusion
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    private ?string $type = null;

    #[ORM\Column(length: 255)]
    private ?string $link = null;

    #[ORM\ManyToOne(inversedBy: 'diffusions')]
    #[ORM\JoinColumn(nullable: false)]
    private ?Iniciative $iniciative = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getType(): ?string
    {
        return $this->type;
    }

    public function setType(string $type): static
    {
        $this->type = $type;

        return $this;
    }

    public function getLink(): ?string
    {
        return $this->link;
    }

    public function setLink(string $link): static
    {
        $this->link = $link;

        return $this;
    }

    public function getIniciative(): ?iniciative
    {
        return $this->iniciative;
    }

    public function setIniciative(?iniciative $iniciative): static
    {
        $this->iniciative = $iniciative;

        return $this;
    }
}

File: ANDROID_4VODS/app/src/main/java/com/javierprado/android_4vods/models/Diffusion.java
package com.javierprado.android_4vods.models;

public class Diffusion {
    private int id;
    private String type;
    private String link;

    public Diffusion(int id, String type, String link) {
        this.id = id;
        this.type = type;
        this.link = link;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getLink() {
        return link;
    }

    public void setLink(String link) {
        this.link = link;
    }
}
